Update raw_connect to simplify validation from UI

gberginc opened the pull request:
The UI repo has changed the way endpoint validation occurs, causing a
problem withinin the angular controller when creating the arguments for
`raw_connect`. Previously, the `auth_url` was required as an argument
which mandated the need to expose the `auth_url` from the Nuage manager
so that it could be used in the UI controller.

Instead, this patch replaces `auth_url` with the specific arguments that
are needed to build the `auth_url` so that the UI can directly pass
those values into the manager.

gberginc commented:
@miq-bot assign @blomquisg 

The purpose of this PR is to offer an alternative to #31. Even though #31 helps the UI to validate the endpoint, I would rather use the approach of most other providers that simply get credentials and endpoint configuration. One of the additional benefits is that the arguments are now stripped (as other providers) because previously we were actually able to break the provider when whitespaces were present in the params.

@blomquisg If you agree with this one, we will close #31 and then also update the UI part.

/cc @gasper-vrhovsek

Fryguy commented on app/models/manageiq/providers/nuage/manager_mixin.rb:
```diff
@@ -2,24 +2,37 @@ module ManageIQ::Providers::Nuage::ManagerMixin
+
+    def auth_url(protocol, server, port, version)
+      scheme = protocol == "ssl-with-validation" ? "https" : "http"
+      "#{scheme}://#{server}:#{port}/nuage/api/#{version}"
```
This should use a URI builder instead (though I see this is a copy, so it can be done in a separate PR).  The main reason is that for IPv6, if an IPv6 server address is given, this code won't actually work...prefer

```ruby
URI::Generic.build(:scheme => scheme, :host => server, :port => port, :path => "/nuage/api/#{version}")
```

gberginc commented:
@blomquisg @Fryguy I have addressed the comment made by @Fryguy regarding the use of `URI::Generic.build`. I've added a spec to make sure URLs did not change.
